bitnami/oauth2-proxy/templates/_helpers.tpl

```diff
@@ -113,8 +113,10 @@ Get the password secret.
 Get the password key to be retrieved from Redis&reg; secret.
 */}}
 {{- define "oauth2-proxy.redis.secretPasswordKey" -}}
-{{- if and .Values.redis.auth.existingSecret .Values.redis.auth.existingSecretPasswordKey -}}
+{{- if and .Values.redis.enabled .Values.redis.auth.existingSecret .Values.redis.auth.existingSecretPasswordKey -}}
 {{- printf "%s" .Values.redis.auth.existingSecretPasswordKey -}}
+{{- else if .Values.externalRedis.existingSecretPasswordKey -}}
+{{- printf "%s" .Values.externalRedis.existingSecretPasswordKey -}}
```

Comment:
This logic may lead to inconsistencies... If `redis` is enabled, the properties under `externalRedis` should be ignored.

Although unlikely, consider the situation in which `redis.enabled=true` but there is no `existingSecret` (the chart should create a new one, which will hold the password in a key named `redis-password`). If `externalRedis.existingSecretPasswordKey` is specified, this will be taken as the key value and will probably lead to error (this is not the same behaviour as before).

Reply:
Completely agree, thanks a lot for the heads up.